include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_fpga.h

```diff
@@ -77,218 +77,35 @@ struct __parallel_for_fpga_submitter<__internal::__optional_kernel_name<_Name...
-
-//------------------------------------------------------------------------
-// parallel_partial_sort
-//-----------------------------------------------------------------------
-
-// TODO: check if it makes sense to move these wrappers out of backend to a common place
-template <typename _ExecutionPolicy, typename _Iterator, typename _Compare,
-          oneapi::dpl::__internal::__enable_if_fpga_execution_policy<_ExecutionPolicy, int> = 0>
-auto
-__parallel_partial_sort(_ExecutionPolicy&& __exec, _Iterator __first, _Iterator __mid, _Iterator __last,
-                        _Compare __comp)
-{
-    // workaround until we implement more performant version for patterns
-    return oneapi::dpl::__par_backend_hetero::__parallel_partial_sort(__exec.__device_policy(), __first, __mid, __last,
-                                                                      __comp);
-}
-
 //------------------------------------------------------------------------
 // parallel_histogram
 //-----------------------------------------------------------------------
 
 // TODO: check if it makes sense to move these wrappers out of backend to a common place
-template <typename _ExecutionPolicy, typename _Event, typename _Range1, typename _Range2, typename _BinHashMgr,
-          oneapi::dpl::__internal::__enable_if_fpga_execution_policy<_ExecutionPolicy, int> = 0>
+template <typename _ExecutionPolicy, typename _Event, typename _Range1, typename _Range2, typename _BinHashMgr>
 auto
-__parallel_histogram(_ExecutionPolicy&& __exec, const _Event& __init_event, _Range1&& __input, _Range2&& __bins,
-                     const _BinHashMgr& __binhash_manager)
+__parallel_histogram(oneapi::dpl::__internal::__fpga_backend_tag, _ExecutionPolicy&& __exec, const _Event& __init_event,
+                     _Range1&& __input, _Range2&& __bins, const _BinHashMgr& __binhash_manager)
 {
-    if constexpr (sizeof(oneapi::dpl::__internal::__value_t<_Range2>) <= sizeof(::std::uint32_t))
-    {
-        // workaround until we implement more performant version for patterns
-        return oneapi::dpl::__par_backend_hetero::__parallel_histogram(
-            __exec.__device_policy(), __init_event, ::std::forward<_Range1>(__input), ::std::forward<_Range2>(__bins),
-            __binhash_manager);
-    }
-    else
-    {
-        static_assert(false, "histogram is not supported on FPGA devices with output types greater than 32 bits");
-        return __future(sycl::event{});
-    }
+    static_assert(sizeof(oneapi::dpl::__internal::__value_t<_Range2>) <= sizeof(::std::uint32_t),
+                  "histogram is not supported on FPGA devices with output types greater than 32 bits");
+
+    // workaround until we implement more performant version for patterns
+    return oneapi::dpl::__par_backend_hetero::__parallel_histogram(
+        oneapi::dpl::__internal::__device_backend_tag{}, __exec, __init_event, ::std::forward<_Range1>(__input),
+        ::std::forward<_Range2>(__bins), __binhash_manager);
```

Comment:
The adjustment away from `static_assert(false,...` is an improvement, but we still do need the constexpr to avoid calling `__parallel_histogram` in this case, (and the alternative return).  Build errors occur for this code within this code if compiled with 64 bit output type.

Reply:
I should clarify, without the constexpr, fpga compiler fails with a difficult to track down error.  I'm not sure if the `static_assert` would be shown first or not, or if it would end up being suppressed.  In any case, I would prefer we continue to avoid calling `__parallel_histogram` with a constexpr in under these conditions.

Reply:
@danhoeflinger I don't quite understood your point. What you mean?
From my point of view, this code absolutely correct and equal to your initial state from functional point of view.
Could you please explain?

Reply:
Yes, I was mistaken.  I tested the branch and as you describe, the build stops at the static assertion with the appropriate message.  
I was worried that the way in which the fpga compiler reported the error would suppress this static assertion, but that is not the case.  You can ignore my previous comments.

Reply:
Thank you, @danhoeflinger !